packages/@react-aria/aria-modal-polyfill/test/index.test.js

```diff
@@ -170,4 +171,173 @@ describe('watchModals', () => {
+            </Menu>
+          </MenuTrigger>
+        </Provider>
+        <hr />
+      </>
+    );
+    expect(() => getByRole('separator')).not.toThrow();
+    act(() => {
+      triggerPress(getByLabelText('Trigger'));
+    });
+    act(() => {
+      jest.runAllTimers();
+    });
+    let menu = getByRole('menu');
+    await waitFor(() => {
+      expect(menu).toBeVisible();
+    });
+    // we shouldn't be able to find it because the search is done by accessibility and the non-modal
+    // part of the tree should be inaccessible while the modal is open
+    expect(() => getByRole('separator')).toThrow();
+
+    expect(document.activeElement).toBe(menu);
+
+    fireEvent.keyDown(menu, {key: 'Escape'});
+    fireEvent.keyUp(menu, {key: 'Escape'});
+
+    act(() => {
+      jest.runAllTimers();
+    });
+    await waitFor(() => {
+      expect(menu).not.toBeInTheDocument();
+    }); // wait for animation
+
+    // once the dialog is removed, we should be able to access the main part of the document again
+    expect(() => getByRole('separator')).not.toThrow();
+  });
+
+  it('should hide around Tray', async () => {
+    let withSection = [
+      {name: 'Heading 1', children: [
+        {name: 'Foo'},
+        {name: 'Bar'},
+        {name: 'Baz'}
+      ]}
+    ];
+    // menu should be a tray
+    matchMedia.useMediaQuery('(max-width: 700px)');
+    watchModals();
+    let {getByLabelText, getByRole} = render(
+      <>
+        <Provider theme={theme}>
+          <MenuTrigger>
+            <ActionButton aria-label="Trigger" />
+            <Menu items={withSection}>
+              {item => (
+                <Section key={item.name} items={item.children} title={item.name}>
+                  {item => <Item key={item.name} childItems={item.children}>{item.name}</Item>}
+                </Section>
+              )}
+            </Menu>
+          </MenuTrigger>
+        </Provider>
+        <hr />
+      </>
+    );
+    expect(() => getByRole('separator')).not.toThrow();
+    act(() => {
+      triggerPress(getByLabelText('Trigger'));
+    });
+    act(() => {
+      jest.runAllTimers();
+    });
+    let menu = getByRole('menu');
+    await waitFor(() => {
+      expect(menu).toBeVisible();
+    });
+    // we shouldn't be able to find it because the search is done by accessibility and the non-modal
+    // part of the tree should be inaccessible while the modal is open
+    expect(() => getByRole('separator')).toThrow();
+
+    expect(document.activeElement).toBe(menu);
+
+    fireEvent.keyDown(menu, {key: 'Escape'});
+    fireEvent.keyUp(menu, {key: 'Escape'});
+
+    act(() => {
+      jest.runAllTimers();
+    });
+    await waitFor(() => {
+      expect(menu).not.toBeInTheDocument();
+    }); // wait for animation
+
+    // once the dialog is removed, we should be able to access the main part of the document again
+    expect(() => getByRole('separator')).not.toThrow();
+  });
+
+  it('should hide around Popover', async () => {
```

Comment:
They day we wrote tests to pass in params to test v2 and v3, could we do that here since these three tests are, menu, menu as tray and dialog, otherwise they are the same.